File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size()-1;
        
        while(l<=r) {
            int x = (l+r)/2;
            if(nums[x]==target) {
                return x;
            }
            else if(nums[x]>target) {
                r=x-1;
            }
            else {
                l=x+1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-1.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] nums = new int[nums1.length + nums2.length];
        int index = 0;
        for(int i = 0; i < nums1.length; i++) {
            nums[index++] = nums1[i];
        }
        for(int i = 0; i < nums2.length; i++) {
            nums[index++] = nums2[i];
        }
        Arrays.sort(nums);

        if(nums.length % 2 == 0) {
            return (nums[nums.length / 2 - 1] + nums[nums.length / 2]) / 2.0;
        }
        else {
            return nums[nums.length / 2];
        }
    }
}
